packages/react-hot-loader/src/reconciler/hotReplacementRender.js

```diff
@@ -168,8 +167,7 @@ const hotReplacementRender = (instance, stack) => {
       // copy over props as long new component may be hidden inside them
       // child does not have all props, as long some of them can be calculated on componentMount.
       const nextProps = { ...instance.props }
-      const childProps = child.props || {}
```

Comment:
I'm not sure if this check was related to the indeterminate component handling, or if it was there for some other reason. Tests seem to pass without it.